### DISCUSSION TITLE
Architecture, Layers and DDD using Mobx

### DISCUSSION THREAD

**Discussion by Joao-vi** [mobxjs/mobx#3857] [General]
Hey there, all good ? 

I'm very interested about how do you architect and design your system using Mobx. I'm looking forward to get some insights as well.

**Comment by Samwell1996** [comment]
Hi!

We usually design our MobX setup around domain entities and normalized data, rather than large nested stores. That keeps updates, async flows, and optimistic changes much easier to reason about.

In general:

- MobX is used as a reactive state layer, not as a place for UI logic.
- We prefer explicit actions / flows over mutating state from components.
- Normalization helps a lot once you deal with caching, offline, or sync issues.

Happy to share more if you have a specific use case in mind 🙂

**Comment by js2me** [comment]
Hello!   
We are using [FSD architectural methodology](https://feature-sliced.design/) + MVVM pattern   
And React works only as view layer, ViewModel classes and Models - (features/entities)
